sunspot/lib/sunspot/type.rb

```diff
@@ -91,8 +91,29 @@ def to_literal(object)
     # available to keyword search.
     #
     class TextType < AbstractType
-      def indexed_name(name) #:nodoc:
-        "#{name}_text"
+      VALID_LANGUAGES = %w[
+        ar bg ca cjk cz da de el en es eu exact fa fi fr ga gl hi hu hy 
+        id it ja phonetic lv nl no pt ro ru sv th tr
+        ]
+        
+      def indexed_name(name, options = {}) #:nodoc:
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Cyclomatic complexity for indexed_name is too high. [8/6]](https://www.codacy.com/app/joni-amir/sunspot/file/2482302916/issues/source?bid=3265104&fileBranchId=3265153#l99)